ecoscope_workflows/compiler.py

```diff
@@ -394,11 +393,14 @@ def _omit_args(self) -> list[str]:
             + [arg for t in self.spec.workflow for arg in t.map_iterable]
         )
 
-    def get_params_jsonschema(self) -> dict[str, dict]:
-        return {
-            t.id: t.known_task.parameters_jsonschema(omit_args=self._omit_args)
+    def get_params_jsonschema(self) -> dict[str, Any]:
+        schema: dict[str, Any] = {}
+        schema["title"] = "Ecoscope Workflow Configurations Form"
+        schema["properties"] = {
+            t.name: t.known_task.parameters_jsonschema(omit_args=self._omit_args)
             for t in self.spec.workflow
         }
+        return schema
```

Comment:
```suggestion
    def get_params_jsonschema(self) -> dict[str, Any]:
        return ReachJSONSchemaForm(
            title="Ecoscope Workflow Configurations Form"
            properties={
                t.name: t.known_task.parameters_jsonschema(omit_args=self._omit_args)
                for t in self.spec.workflow
            },
        ).model_dump()
```

> The challenge in reusing ReactJSONSchemaFormFilters is the absence of RJSFFilter. I considered the same approach initially, but currently, the JSON schema for each task is generated by TypeAdapter of a func. Thus, we have to create a data model for each task, which will break auto-generation of configuration from a function signature. Any thoughts?


Something like this diff ☝️ is what I was imagining. It would require some generalization of the RJSF* models...

Reply:
These models 

https://github.com/wildlife-dynamics/ecoscope-workflows/blob/4ed2ec81f785fcd1d00294fb47679f131dadfd71/ecoscope_workflows/jsonschema.py#L66-L74

could be generalized something like ...

```diff
-class RJSFFilter(BaseModel): 
-   """Model representing a React JSON Schema Form filter.""" 
+class RJSFField(BaseModel): 
+   """Model representing a React JSON Schema Form.""" 

-   property: RJSFFilterProperty
-   uiSchema: RJSFFilterUiSchema
+   property: RJSFFilterProperty | dict
+   uiSchema: RJSFFilterUiSchema | dict


-class ReactJSONSchemaFormFilters(BaseModel): 
-    options: dict[str, RJSFFilter] 
+class ReactJSONSchemaFormFilters(BaseModel):
+    title: str
+    properties: dict[str, RJSFField] 

```

This is just some quick pseudocode, happy to sync more on the details if that would be useful.

Reply:
Thanks. I like this!